Gordon360/AuthorizationFilters/StateYourBusiness.cs

```diff
@@ -503,6 +550,32 @@ private bool canUpdate(string resource)
                     }
                 case Resource.STUDENT:
                     return false; // No one should be able to update a student through this API
+                case Resource.HOUSING:
+                    {
+                        // The housing admins can update the application information (i.e. probation, offcampus program, etc.)
+                        // If the user is a student, then the user must be on an application and be an editor to update the application
+                        var housingService = new HousingService(new UnitOfWork());
+                        if (user_position == Position.SUPERADMIN || (housingService.CheckIfHousingAdmin(user_id)))
+                        {
+                            return true;
+                        }
+                        else if (user_position == Position.STUDENT)
+                        {
+                            var sess_cde = Helpers.GetCurrentSession().ToString();
+                            int? applicationID = housingService.GetApplicationID(user_name, sess_cde);
+                            var requestedApplicationID = (int)context.ActionArguments["applicationID"];
+                            if (applicationID.HasValue && applicationID == requestedApplicationID)
+                            {
+                                var editorUsername = housingService.GetEditorUsername((int)applicationID);
+                                var is_editor = editorUsername == user_name;
+                                if (is_editor)
```

Comment:
You could probably combine these lines, since this variable is calculated and then used only once directly afterwards:
```suggestion
                                if (editorUsername == user_name)
```